### PR TITLE
[Bug]: show full tables from XXX Error (#5559)

### PR DESCRIPTION
**Pull request by badboynt1**
## What type of PR is this?

- [ ] API-change
- [X] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5559

## What this PR does / why we need it:
support show full tables statement

### CODE REVIEW

**Review by aunjgr — CHANGES_REQUESTED** [review]
Please add bvt cases to verify your fix.

**Review by aunjgr — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/plan/build_show.go [hunk 1]
```diff
@@ -246,7 +246,11 @@ func buildShowTables(stmt *tree.ShowTables, ctx CompilerContext) (*Plan, error)
 		return nil, moerr.NewNoDB()
 	}
 	ddlType := plan.DataDefinition_SHOW_TABLES
-	sql := fmt.Sprintf("SELECT relname as Tables_in_%s FROM %s.mo_tables WHERE reldatabase = '%s' and relname != '%s'", dbName, MO_CATALOG_DB_NAME, dbName, "%!%mo_increment_columns")
+	var tableType string
+	if stmt.Full {
+		tableType = ", case relkind when 'v' then 'VIEW' else 'BASE TABLE' end as Table_type"
+	}
+	sql := fmt.Sprintf("SELECT relname as Tables_in_%s %s FROM %s.mo_tables WHERE reldatabase = '%s' and relname != '%s'", dbName, tableType, MO_CATALOG_DB_NAME, dbName, "%!%mo_increment_columns")
```
Comment (fengttt) [inline pkg/sql/plan/build_show.go:253]: This is subject to sql injection.

Comment (badboynt1) [reply pkg/sql/plan/build_show.go:253]: > This is subject to sql injection.

bvt tests about this added. no sql injection problem here

Comment (fengttt) [reply pkg/sql/plan/build_show.go:253]: Of course there is sql injection here.   You are just not creative enough to come up with one.

There must be a quote_identify call, or using param binding.

Comment (badboynt1) [reply pkg/sql/plan/build_show.go:253]: this patch is to support show full tables statement. 
sql injection problem will be fixed in https://github.com/matrixorigin/matrixone/issues/5476

**Review by fengttt — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**